code/modules/mob/living/carbon/human/species_types/diona.dm

```diff
@@ -0,0 +1,234 @@
+	mutant_bodyparts = list("diona_leaves", "diona_thorns", "diona_flowers", "diona_moss", "diona_mushroom", "diona_antennae", "diona_eyes", "diona_pbody")
+	default_features = list("diona_leaves" = "None", "diona_thorns" = "None", "diona_flowers" = "None", "diona_moss" = "None", "diona_mushroom" = "None", "diona_antennae" = "None", "body_size" = "Normal", "diona_eyes" = "None", "diona_pbody" = "None")
+	inherent_factions = list("plants", "vines")
+	fixed_mut_color = "59CE00"
+	attack_verb = "slash"
+	attack_sound = 'sound/weapons/slice.ogg'
+	miss_sound = 'sound/weapons/slashmiss.ogg'
+	burnmod = 1.25
+	heatmod = 1.5
+	brutemod = 0.8
+	staminamod = 0.7
+	meat = /obj/item/food/meat/slab/human/mutant/diona
+	exotic_blood = /datum/reagent/water
+	species_gibs = GIB_TYPE_ROBOTIC //Someone please make this like, xeno gibs or something in the future. I cant be bothered to fuck around with gib code right now.
+	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_MAGIC | MIRROR_PRIDE | ERT_SPAWN | RACE_SWAP
+	species_language_holder = /datum/language_holder/diona
+	bodytemp_normal = (BODYTEMP_NORMAL - 22) // Body temperature for dionae is much lower then humans as they are plants, supposed to be 15 celsius
+	speedmod = 2 // Dionae are slow.
+	species_height = SPECIES_HEIGHTS(-1, 0, -2) //Naturally tall.
+	swimming_component = /datum/component/swimming/diona
+	inert_mutation = DRONE
+
+	mutanteyes = /obj/item/organ/eyes/diona //SS14 sprite
+	mutanttongue = /obj/item/organ/tongue/diona //Dungeon's sprite
+	mutant_brain = /obj/item/organ/brain/diona //SS14 sprite
+	mutantliver = /obj/item/organ/liver/diona //Dungeon's sprite
+	mutantlungs = /obj/item/organ/lungs/diona //Dungeon's sprite
+	mutantstomach = /obj/item/organ/stomach/diona //SS14 sprite
+	mutantears = /obj/item/organ/ears/diona //SS14 sprite
+	mutant_heart = /obj/item/organ/heart/diona //Dungeon's sprite
+	mutant_organs = list()
+
+	species_chest = /obj/item/bodypart/chest/diona
+	species_head = /obj/item/bodypart/head/diona
+	species_l_arm = /obj/item/bodypart/l_arm/diona
+	species_r_arm = /obj/item/bodypart/r_arm/diona
+	species_l_leg = /obj/item/bodypart/l_leg/diona
+	species_r_leg = /obj/item/bodypart/r_leg/diona
+
+	var/datum/action/diona/split/split_ability //All dionae start with this.
+	var/mob/living/simple_animal/nymph/drone = null
+
+/datum/species/diona/spec_life(mob/living/carbon/human/H)
+	if(H.stat == DEAD)
+		return
+	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
+	if(isturf(H.loc)) //else, there's considered to be no light
+		var/turf/T = H.loc
+		light_amount = min(1,T.get_lumcount()) - 0.5
+		H.adjust_nutrition(light_amount * 10)
+		if(H.nutrition > NUTRITION_LEVEL_ALMOST_FULL)
+			H.set_nutrition(NUTRITION_LEVEL_ALMOST_FULL)
+		if(light_amount > 0.2) //if there's enough light, heal
+			H.heal_overall_damage(1,1, 0, BODYTYPE_ORGANIC)
+			H.adjustToxLoss(-1)
+			H.adjustOxyLoss(-1)
```

Comment:
This seems like exceptionally strong healing. If they are in crit they should also stop healing. This would be preferable as some kind of status effect which shows why they are getting healed, and should require them to remain in bright light for a long period of time.

Alternatively, the healing should only work on minor injuries (<20 damage)